recipe/meta.yaml

```diff
@@ -43,61 +44,149 @@ requirements:
-    # by run_exports, run_exports doesn't necessary
-    # bring in compile time depenencies
-    - zlib
 
-test:
-  requires:
-    - glib  # needed for the pkg-config invocations below
-    - pkg-config
+outputs:
+
+  # Everything needed to build against libcairo
+  - name: cairo
+    script: install.py
+    build:
+      run_exports:
+        - {{ pin_subpackage('libcairo' ~ so_name_major) }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - pthread-stubs         # [linux]
+      host:
+        - zlib
+        - {{ pin_subpackage('libcairo' ~ so_name_major, exact=True) }}
+      run:
+        - {{ pin_subpackage('libcairo' ~ so_name_major, exact=True) }}
+        # hmaarrfk: 2022/08
+        # While the dependencies below should be captured
+        # by run_exports, run_exports doesn't necessary
+        # bring in compile time depenencies
+        - zlib
+    test:
+      requires:
+        - glib  # needed for the pkg-config invocations below
+        - pkg-config
+
+      commands:
+        # Check commands.
+        - cairo-trace --help  # [not win]
+
+        # Verify libraries. Note that on Windows, we patch Cairo's build system to
+        # produce DLL names like `cairo.dll` rather than `cairo-2.dll` to maintain
+        # continuity with binaries shipped before the switch to Meson.
+        {% set cairo_libs = [
+                "cairo",
+                "cairo-gobject",
+                "cairo-script-interpreter",
+        ] %}
+        {% for each_cairo_lib in cairo_libs %}
+        - test ! -f $PREFIX/lib/lib{{ each_cairo_lib }}.a                   # [not win]
+        - test   -f $PREFIX/lib/lib{{ each_cairo_lib }}.dylib               # [osx]
+        - test   -f $PREFIX/lib/lib{{ each_cairo_lib }}.so                  # [linux]
+        - if not exist %LIBRARY_LIB%\{{ each_cairo_lib }}.lib exit /b 1   # [win]
+        - if     exist %LIBRARY_LIB%\lib{{ each_cairo_lib }}.a exit /b 1  # [win]
+        {% endfor %}
+
+        # Check pkg-config files.
+        - test -f $PREFIX/lib/pkgconfig/cairo-quartz.pc                  # [osx]
+        - test -f $PREFIX/lib/pkgconfig/cairo-xlib.pc                    # [linux]
+        - if not exist %LIBRARY_LIB%\pkgconfig\cairo-win32.pc exit /b 1  # [win]
+
+        # check include files
+        - test -f $PREFIX/include/cairo/cairo.h               # [unix]
+        - if not exist %LIBRARY_INC%\cairo\cairo.h exit /b 1  # [win]
+
+        # check that cairo was built with fontconfig support
+        - grep -q "CAIRO_HAS_FC_FONT 1" $PREFIX/include/cairo/cairo-features.h  # [unix]
+
+        - export PKG_CONFIG_PATH=$PREFIX/lib/pkgconfig   # [not win]
+        - set "PKG_CONFIG_PATH=%LIBRARY_LIB%\pkgconfig"  # [win]
+        - pkg-config --cflags cairo
+        - pkg-config --cflags cairo-fc
+        - pkg-config --cflags cairo-gobject
+
+  # Only what's needed for run_exports downstream
+  - name: libcairo{{ so_name_major }}
```

Comment:
If it isn't co-installable, why include the major version in the name?

Reply:
I assume you are referring to the Windows build variant because the unix packages would be coinstallable with other versions of different ABI major version.

The reason is to keep consistency between Windows and Unix package names.